go/libraries/doltcore/sqle/stats/auto_refresh.go

```diff
@@ -0,0 +1,303 @@
+					continue
+				}
+
+				for _, table := range tables {
+					sqlTable, ok, err := sqlDb.GetTableInsensitive(sqlCtx, table)
+					if err != nil {
+						sqlCtx.GetLogger().Debugf("statistics refresh error: %s", err.Error())
+						continue
+					}
+					if !ok {
+						sqlCtx.GetLogger().Debugf("statistics refresh error: table not found %s", table)
+						continue
+					}
+
+					var dTab *doltdb.Table
+					switch t := sqlTable.(type) {
+					case *sqle.AlterableDoltTable:
+						dTab, err = t.DoltTable.DoltTable(sqlCtx)
+					case *sqle.WritableDoltTable:
+						dTab, err = t.DoltTable.DoltTable(sqlCtx)
+					case *sqle.DoltTable:
+						dTab, err = t.DoltTable(sqlCtx)
+					default:
+						err = fmt.Errorf("failed to unwrap dolt table from type: %T", sqlTable)
+					}
+					if err != nil {
+						sqlCtx.GetLogger().Debugf("statistics refresh error: %s", err.Error())
+						continue
+					}
+
+					tableHash, err := dTab.GetRowDataHash(ctx)
+					if err != nil {
+						sqlCtx.GetLogger().Debugf("statistics refresh error: %s", err.Error())
+						continue
+					}
+
+					if curStats.latestTableHashes[table] == tableHash {
+						// no data changes since last check
+						tableExistsAndSkipped[table] = true
+						sqlCtx.GetLogger().Debugf("statistics refresh: table hash unchanged since last check: %s", tableHash)
+						continue
+					} else {
+						sqlCtx.GetLogger().Debugf("statistics refresh: new table hash: %s", tableHash)
+					}
+
+					iat, ok := sqlTable.(sql.IndexAddressableTable)
+					if !ok {
+						sqlCtx.GetLogger().Debugf("statistics refresh error: table does not support indexes %s", table)
+						continue
+					}
+
+					indexes, err := iat.GetIndexes(sqlCtx)
+					if err != nil {
+						sqlCtx.GetLogger().Debugf("statistics refresh error: %s", err.Error())
+						continue
+					}
+
+					// collect indexes and ranges to be updated
+					var idxMetas []indexMeta
+					for _, index := range indexes {
+						qual := sql.NewStatQualifier(dbName, table, strings.ToLower(index.ID()))
+						qualExists[qual] = true
+						curStat := curStats.stats[qual]
+						if curStat == nil {
+							curStat = NewDoltStats()
+							curStat.Qual = qual
+
+							cols := make([]string, len(index.Expressions()))
+							tablePrefix := fmt.Sprintf("%s.", table)
+							for i, c := range index.Expressions() {
+								cols[i] = strings.TrimPrefix(strings.ToLower(c), tablePrefix)
+							}
+							curStat.Columns = cols
+						}
+						sqlCtx.GetLogger().Debugf("statistics refresh index: %s", qual.String())
+
+						idxMeta, err := newIdxMeta(sqlCtx, curStat, dTab, index, curStat.Columns)
+						if err != nil {
+							sqlCtx.GetLogger().Debugf("statistics refresh error: %s", err.Error())
+							continue
+						}
+						curCnt := float64(len(curStat.active))
+						updateCnt := float64(len(idxMeta.updateChunks))
+						deleteCnt := float64(len(curStat.active) - len(idxMeta.preexisting))
```

Comment:
Should this be reversed? The number deleted is the pre-existing minus the current count?

Reply:
the naming here is somewhat rough and ready, but `curStat` is the existing statistics on disk, `idxMeta` is the update metadata. `idxMeta.preexisting` is the fraction of chunks in the update metadata overlapping with `curStat`'s chunks. Will change `idxMeta` to indicate it's a candidate update